library/arith.lisp

```diff
@@ -480,6 +493,136 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
+
+  (declare sincos (Trigonometric :a => :a -> (Tuple :a :a)))
+  (define (sincos x)
+    (tuple (sin x) (cos x)))
+
+  (define-class (Exponentiable :a)
+    (**      (:a -> :a -> :a))
+    (logBase (:a -> :a -> :a)))
+
+  (declare sqrt ((Dividable Integer :a) (Exponentiable :a) => :a -> :a))
+  (define (sqrt x)
+    "Returns the square root of the input"
+    (** x (general/ (the integer 1) (the integer 2))))
+
+  (define-class ((Num :a) (Dividable :a :a) (Trigonometric :a) (Exponentiable :a) => Float :a)
+    (ee :a)
+    (pi :a))
+
+  (declare exp ((Float :f) => :f -> :f))
+  (define (exp x)
+    "Raise ee to the power of X"
+    (** ee x))
+
+  (declare log ((Float :f) => :f -> :f))
+  (define (log x)
+    "The natural logarithm of a given value"
+    (logBase ee x))
+
+  (define-class ((Float :f) (Ord :f) => RealFloat :f)
+    "Real Floating-Point Numbers"
+    (rationalize (:f -> Fraction)))
+
+  (declare atan2 ((RealFloat :f) => :f -> :f -> :f))
+  (define (atan2 y x)
+    "computes the arc tangent of y/x,
```

Comment:
Write in capitalized and complete sentences, no line breaks.

> Computes the two-argument arctangent of y and x, which is roughly the same as `(atan (/ y x))` when defined and accounting for the quadrant of the point (x, y).

Reply:
Fixed